### PR TITLE
Poo#18850 firefox java support test

### PR DESCRIPTION
**Pull request by rwx788**
In one of the releases java was not supported in firefox. We already have such regression test for sle, which was migrated to TW and Leap.
This commit includes needles required to the test for TW and Leap 42.3 on gnome.
As per ticket suggestion, moved tests from x11regressions/firefox/sle12 to tests/x11regressions/firefox in order to exclude specific version in the path name.

Progress reference: https://progress.opensuse.org/issues/18850
Verification run Leap 42.3: http://gershwin.arch.suse.de/tests/102
Verification run TW: http://gershwin.arch.suse.de/tests/101

Has to be merged along with needles PR: https://github.com/os-autoinst/os-autoinst-needles-opensuse/pull/207

### CODE REVIEW

File: lib/main_common.pm [hunk 1]
```diff
@@ -455,6 +455,8 @@ sub load_extra_tests() {
                     loadtest "x11/gdm_session_switch";
                 }
                 loadtest "x11/seahorse";
+                # poo#18850 java test support for firefox
```
Comment (okurz) [inline lib/main_common.pm]: schedule `tests/x11regressions/x11regressions_setup.pm` here, see below for explanation

File: lib/x11regressiontest.pm [hunk 2]
```diff
@@ -454,9 +454,13 @@ sub post_fail_hook() {
 }
 
 sub start_firefox() {
+    my ($chown_ser_dev) = shift;
     mouse_hide(1);
 
     x11_start_program 'xterm';
+    if ($chown_ser_dev) {
+        testapi::assert_script_sudo("chown $testapi::username /dev/$testapi::serialdev");
+    }
```
Comment (okurz) [inline lib/x11regressiontest.pm]: the `chown` on the serialdev should probably be called by scheduling `tests/x11regressions/x11regressions_setup.pm` unconditionally.

File: tests/x11regressions/firefox/firefox_java.pm [hunk 3]
```diff
@@ -53,13 +53,21 @@ sub java_testing {
 
 sub run() {
     my ($self) = @_;
-    $self->start_firefox;
-
-    assert_and_click('firefox-logo');
+    # Need to change owner of serial devide on opensuse
```
Comment (okurz) [inline tests/x11regressions/firefox/firefox_java.pm]: s/devide/device/

File: tests/x11regressions/firefox/firefox_java.pm [hunk 4]
```diff
@@ -53,13 +53,21 @@ sub java_testing {
...
+    $self->start_firefox(check_var('DISTRI', 'opensuse'));
+    #Required only on sle, as open mozilla home page
```
Comment (okurz) [inline tests/x11regressions/firefox/firefox_java.pm]: this sentence somehow does not seem to make sense, "as open mozilla home page"? missing something?

Comment (rwx788) [inline tests/x11regressions/firefox/firefox_java.pm]: Somehow start_firefox method for opensuse only starts the browser, and home page is not set to mozilla home page, but to search page. I'm not sure why this check is there for sle, as not related to java test at all.

Comment (okurz) [inline tests/x11regressions/firefox/firefox_java.pm]: sure, the home page differs for SLE and openSUSE

Comment (rwx788) [inline tests/x11regressions/firefox/firefox_java.pm]: So, should we remove this firefox-logo click then for sake of consistency?

File: lib/x11regressiontest.pm (hunk 2 above)
Comment (rwx788) [inline lib/x11regressiontest.pm]: Good point, will clean this up. Cheers!

File: tests/x11regressions/firefox/firefox_java.pm (hunk 4 above)
Comment (rwx788) [inline tests/x11regressions/firefox/firefox_java.pm]: This is only remaining open point, verification run here: http://gershwin.arch.suse.de/tests/110#

**Comment by nilxam** [comment]
it does not works on 42.3 https://openqa.opensuse.org/tests/421911#step/firefox_java/5

**Comment by rwx788** [comment]
@nilxam hadn't had this wizard screen in my local run. Fix is on it's way. Thanks!

**Comment by nilxam** [comment]
I think it might relate to the ordering of chrome/firefox, chrome has executed before firefox  on extra_tests_on_gnome scenario, thus firefox sees another browser in the SUT.

**Comment by rwx788** [comment]
@nilxam Yes, reordering is easiest solution. Need to check that it doesn't affect chrome tests. Otherwise, I'll add steps to skip import wizard.

**Comment by rwx788** [comment]
@nilxam Fix submitted here: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/3091 Thanks for being proactive!

**Pull request merged**